src/ServiceControl/SagaAudit/SagaDetailsIndex.cs

```diff
@@ -0,0 +1,58 @@
+namespace ServiceControl.SagaAudit
+{
+    using System.Linq;
+    using Raven.Client.Indexes;
+
+    public class SagaDetailsIndex : AbstractMultiMapIndexCreationTask<SagaHistory>
+    {
+        public SagaDetailsIndex()
+        {
+            AddMap<SagaSnapshot>(docs => from doc in docs
+                select new
+                {
+                    doc.SagaId,
+                    Id = doc.SagaId,
+                    doc.SagaType,
+                    Changes = new[]
+                    {
+                        new SagaStateChange
+                        {
+                            Endpoint = doc.Endpoint,
+                            FinishTime = doc.FinishTime,
+                            InitiatingMessage = doc.InitiatingMessage,
+                            OutgoingMessages = doc.OutgoingMessages,
+                            StartTime = doc.StartTime,
+                            StateAfterChange = doc.StateAfterChange,
+                            Status = doc.Status
+                        }
+                    }
+                });
+
+            //Legacy so we still scan old sagahistories
+            AddMap<SagaHistory>(docs => from doc in docs
```

Comment:
So the old index will still be running. Technically we haven't solved anything when this one is not deprecated or do I misunderstand?